docs/docsite/rst/roadmap/ROADMAP_2_5.rst

```diff
@@ -0,0 +1,217 @@
+
+Runtime Check on Modules for Blacklisting
+-----------------------------------------
+- Filter on things like "supported_by" in module metadata
+- Provide users with an option of "warning, error or allow/ignore"
+- Configurable via ansible.cfg and environment variable
+
+Windows
+-------
+- Implement gather_subset on Windows facts 
+- Move setup.ps1 guts to module_utils to allow arbitrary modules to call/refresh individual facts.
+- Fix Windows binary module support to work properly with become/env/async in all cases.
+- Fix Windows async + become to allow them to work together
+- Solve Windows become/env support for raw/script
+- Implement Windows become flags for controlling various modes
+
+  - logontype
+  - elevation behavior
+  - Add dict support to become_flags.
+- Fix Windows auto-kinit with threaded workers (or disallow it)
+- Finish C#/Powershell module_utils rewrite, convert core modules to use it.
+- Convert win_updates to action plugin for auto reboot and extra features
+- Spike out support for Windows Nano Server
+- Spike out changing the connection over to PSRP instead of WSMV
+- Module updates
+
+  - win_updates
+  
+    - Fix win_updates to detect (or request) become
+    - Add whitelist/blacklist features to win_updates
+  - win_dsc further improvements
+
+General Cloud
+-------------
+- Make multi-cloud provisioning easier
+
+  - Document multi-instance provisioning with loop directive
+  - Extend async_status to accept a list or build new action to simplify the with_items/register/until:finish patterns.
+- Diff mode will output provisioning task results of ansible-playbook runs
+- Terraform module
+
+AWS
+---
+- Focus on pull requests for various modules
+- Triage existing merges for modules
+- Module work
+
+  - ec2_instance 
+  - ec2_vpc: Allow the addition of secondary IPv4 CIDRS to existing VPCs.
+  - AWS Network Load Balancer support (NLB module, ASG support, etc)
+  - rds_instance
+
+Azure
+-----
+- Azure CLI auth
+- Fix Azure module results to have "high-level" output instead of raw REST API dictionary
+- Deprecate Azure automatic storage accounts in azure_rm_virtualmachine
+
+Network Roadmap
+---------------
+- Refactor common network shared code into package
+- Convert various nxos modules to leverage declarative intent 
+- Refactor various modules to leverage the cliconf plugin
+- Add various missing declarative modules for supported platforms and functions
+- Implement a feature that handles platform differences and feature unavailability.
+- netconf-config.py should provide control for deployment strategy
+- Create netconf connection plugin
+- Create netconf fact module
+- Turn network_cli into a usable connection type
+- Implements jsonrpc message passing for ansible-connection
+- Improve logging for ansible-connection
+- Improve stdout output for failures whilst using persistent connection
+- Create IOS-XR NetConf Plugin
+- Refactor junos modules to use netconf plugin
+
```

Comment:
Filters: Add a filter to convert XML response from the network device to JSON object.